sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt

```diff
@@ -96,18 +96,20 @@ abstract class SigstoreSignExtension(private val project: Project) {
         }
         publication.whenPublishableArtifactRemoved {
             val publishableArtifact = this
-            signTask.configure {
-                signatures.findByName(file.name)
-                    ?.takeIf { publishableArtifact in it.builtBy  }
-                    ?.let {
-                        signatures.remove(it)
-                        return@configure
-                    }
-                // Slow path just in case
-                signatures.removeIf { publishableArtifact in it.builtBy }
+            // Ignore artifacts that we have not added a signature for
+            artifacts.remove(publishableArtifact)?.also {
+                signTask.configure {
+                    signatures.findByName(publishableArtifact.file.name)
+                        ?.takeIf { publishableArtifact in it.builtBy  }
+                        ?.let {
+                            signatures.remove(it)
+                            return@configure
+                        }
+                    // Slow path just in case
+                    signatures.removeIf { publishableArtifact in it.builtBy }
+                }
+                publication.removeDerivedArtifact(it)
```

Comment:
```suggestion
            val artifact = artifacts.remove(publishableArtifact) ?: return@whenPublishableArtifactRemoved
            publication.removeDerivedArtifact(artifact)
            signTask.configure {
                signatures.findByName(file.name)
                    ?.takeIf { publishableArtifact in it.builtBy  }
                    ?.let {
                        signatures.remove(it)
                        return@configure
                    }
                // Slow path just in case
                signatures.removeIf { publishableArtifact in it.builtBy }
            }
```

Reply:
I am not a proficient Kotlin user. But from a personal perspective, I find following the flow with different `return@` a bit harder.

Do you have a strong preference for your version @vlsi ?

Reply:
Early exit allows reducing the nesting of the blocks, and it enables easier reasoning.

In this case, we check the artifact, and skip everything if the artifact is unknown. The code like `if (..) { if (...) { ...} } else {... }` is hard to read.

Reply:
I had an `if` based version at first and I agree that the less likable of all. The `also` method feels more natural to me. But that's a style thing effectively.
Will move to your proposal.